Fix CI failures with Python 2.7

JeanChristopheMorinPerso opened the pull request:
Fixes #1502

This PR fixes the issues with Python 2.7 in CI. We now use conda to get and install Python 2.7. This is because GitHub decided that it was alright to completely delete their Python 2.7 interpreters from the internet and break the entire world. The funny thing is that they still compile and ship 3.1 and up :shrug:.

~Note that the csh tests are failing for some reasons. I'm really not sure why. I'm tempted to say that we could ditch csh entirely...~

Oh, and I also updated our actions to use more recent versions.

JeanChristopheMorinPerso commented:
@Jawabiscuit are you of failures on the csh shell tests? You've been running the tests I guess, so maybe you know?

maxnbk commented:
I wouldn't be so quick to throw out csh. Autodesk Flame systems default to it, and I know of environments where rez participates in Flame environments.

Jawabiscuit commented:
@JeanChristopheMorinPerso, csh hasn't posed any problems at all in my testing.

Jawabiscuit commented on .github/actions/setup-python/action.yaml:
```diff
@@ -0,0 +1,60 @@
+name: Setup python
+
+description: Setup python using either actions/setup-pythono or conda
```
```suggestion
description: Setup python using either actions/setup-python or conda
```

JeanChristopheMorinPerso commented:
Alright, I fixed the csh issue. It wasn't related to the tests. I fixed it by simply not using fancy action to install conda. Conda already comes pre-installed on the runners so we don't need to install it. I think the action was injecting stuff into the job environment.

maxnbk reviewed (approved):
Nice work!

JeanChristopheMorinPerso commented:
Considering that this is only affecting the CI, I'll go ahead and merge with a single approval.
